src/coreComponents/physicsSolvers/LinearSolverParameters.cpp

```diff
@@ -20,8 +20,79 @@
 
 namespace geos
 {
+
 using namespace dataRepository;
 
+/**
+ * @brief Register an input block handler for a struct.
+ * @tparam CHILD type of child Group that will handle input for @p BLOCK
+ * @tparam T type for which input handling is needed
+ * @param parent pointer to parent group
+ * @param key the group key (XML tag) for sub-block
+ * @param data the struct to handle input for
+ * @note blocks are assumed optional for now, but it may be changed,
+ *       in which case InputFlags should be provided as a parameter.
+ */
+template< typename CHILD, typename T >
+void registerInputBlock( Group * parent, char const * const key, T & data )
+{
+  parent->registerGroup( key, std::make_unique< CHILD >( key, parent, data ) ).setInputFlags( InputFlags::OPTIONAL );
+}
+
+class BlockParametersInput final : public dataRepository::Group
+{
+public:
+
+  /// Constructor
+  BlockParametersInput( string const & name,
+                        Group * const parent,
+                        LinearSolverParameters::Block & params );
+
+  virtual Group * createChild( string const & childKey, string const & childName ) override
+  {
+    GEOS_UNUSED_VAR( childKey, childName );
+    return nullptr;
+  }
+
+  /// Keys appearing in XML
+  struct viewKeyStruct
+  {
+    static constexpr char const * shapeString()  { return "shape"; }
+    static constexpr char const * schurTypeString() { return "schurType"; }
+    static constexpr char const * scalingString()  { return "scaling"; }
+  };
+
+private:
+
+  LinearSolverParameters::Block & m_parameters;
+};
+
+BlockParametersInput::BlockParametersInput( string const & name,
+                                            Group * const parent,
+                                            LinearSolverParameters::Block & params )
+  :
+  Group( name, parent ),
+  m_parameters( params )
+{
+  registerWrapper( viewKeyStruct::shapeString(), &m_parameters.shape ).
+    setDefaultValue( m_parameters.shape ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Block preconditioner shape, options: "
+                    "``" + EnumStrings< LinearSolverParameters::Block::Shape >::concat( "``, ``" ) + "``" );
+
+  registerWrapper( viewKeyStruct::schurTypeString(), &m_parameters.schurType ).
+    setDefaultValue( m_parameters.schurType ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Schur complement type, options: "
+                    "``" + EnumStrings< LinearSolverParameters::Block::SchurType >::concat( "``, ``" ) + "``" );
+
+  registerWrapper( viewKeyStruct::scalingString(), &m_parameters.scaling ).
+    setDefaultValue( m_parameters.scaling ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Block scaling type, options: "
+                    "``" + EnumStrings< LinearSolverParameters::Block::Scaling >::concat( "``, ``" ) + "``" );
+}
```

Comment:
This is a new way of structuring linear solver parameter inputs: instead of an ever-growing list of prefixed attributes like 
```xml
<LinearSolverParameters
  blockShape="..."
  blockShurType="..."
  blockScaling="..."/>
```
we will start adding nested structure as follows:
```xml
<LinearSolverParameters>
  <Block
    shape="..."
    schurType="..."
    scaling="..."/>
</LinearSolverParameters>
```

This improves readability by indenting parameter blocks according to their logical structure and avoiding repetition of the prefixes. I use this extensively in multiscale work, but will also propose a similar refactoring for AMG parameters.

Reply:
This is very nice, specially if we want to expose some MGR options per level...